optaplanner-core/src/main/java/org/optaplanner/core/api/solver/SolverFactory.java

```diff
@@ -286,7 +271,9 @@
      * @param <Solution_> the solution type, the class with the {@link PlanningSolution} annotation
      */
     public static <Solution_> SolverFactory<Solution_> createEmptyFromKieContainer(KieContainer kieContainer) {
-        return new DefaultSolverFactory<>(new SolverConfig(), new SolverConfigContext(kieContainer));
+        SolverConfig solverConfig = new SolverConfig(kieContainer.getClassLoader());
+        return new DefaultSolverFactory<>(solverConfig,
+                new SolverConfigContext(kieContainer));
```

Comment:
Strange formatting. In my opinion this:
```java
return new DefaultSolverFactory<>(solverConfig, new SolverConfigContext(kieContainer));
```
or this
```java
return new DefaultSolverFactory<>(
        solverConfig,
        new SolverConfigContext(kieContainer)
);
```
are more clear.

Reply:
Adjusted it here.
There are many more cases where this happens (even though we do try to avoid it). None any more in SolverFactory though.